### PR TITLE
[285] Update docs for adding new resources

### PR DESCRIPTION
**Pull request by AbigailMcP**
## Changes made in this PR

 - Split up README into new sections for devs and contributors
 - Add doc for Adding a new resource
 - Add example-resource.md file
 - Link to example PR

https://trello.com/c/obcz4Lqt/285-create-instructions-for-content-designers-and-devs-to-create-and-approve-pull-requests-for-the-service

### CODE REVIEW

File: docs/creating-a-pull-request.md
Comment (AbigailMcP) [inline docs/creating-a-pull-request.md:1]: This is just a straight copy of what was in the original README.

File: docs/set-up.md
Comment (AbigailMcP) [inline docs/set-up.md:1]: Again, copied straight from the README, no changes.

**Comment by AbigailMcP** [comment]
[create instructions for content designers and devs to create and approve pull requests for the service](https://trello.com/c/obcz4Lqt/285-create-instructions-for-content-designers-and-devs-to-create-and-approve-pull-requests-for-the-service)

File: README.md [hunk 1]
```diff
@@ -1,197 +1,33 @@
-# Improve School Workload and Wellbeing
+# Improve school workload and wellbeing
```
Comment (claire-hughez) [inline README.md:1]: It should now be 'Improve workload and wellbeing for school staff'

we had to change names just before you joined Abi.

File: docs/adding-a-resource.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
+# Adding a new resource
+
+All resources are written using Markdown. Markdown files are stored in the
```
Comment (claire-hughez) [inline docs/adding-a-resource.md:3]: Should we specify here that there may be HTML on the page for any boxes or more complicated parts on the page?

File: docs/adding-a-resource.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+`content` folder. This is where you'll find or create new `.md` files for
```
Comment (claire-hughez) [inline docs/adding-a-resource.md:4]: .md does that just mean It's a markdown file?

File: docs/adding-a-resource.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+resources.
+
+Useful links:
+
+- [Example resource file](docs/example-resource.md)
+
+- [Example PR](https://github.com/DFE-Digital/improve-school-workload-and-wellbeing/pull/39)
```
Comment (claire-hughez) [inline docs/adding-a-resource.md:11]: I think some people might not realise why an example PR is useful.

Maybe:
This is an example pull request. This is how changes happen. If you copy the changes in this pull request, you can add a new resource. (or something less lame)

File: docs/adding-a-resource.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+## 2. Add your resource Markdown
+
+Using the code in `example-resource.md` as a guide, copy the Markdown for your
+resource into your new file.
+
+## 3. Add your resource HTML
+
+Due to their complexity, some elements of your Markdown page will need to be
+written inline in HTML. The following code snippets are used in
```
Comment (claire-hughez) [inline docs/adding-a-resource.md:32]: what does inline mean?

File: docs/adding-a-resource.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+corresponding explore-all-resources file.
+
+e.g. if you are creating the 'Establish a wellbeing committee' resource and this
+is a staff wellbeing topic, open `content/explore-all-resources/staff-wellbeing.html.erb`.
+
+Add your new resource's URL to the correct render block:
+
+```ruby
+<%= render('/partials/explore_resource_card.*',
+  title: "Establishing a wellbeing committee",
```
Comment (claire-hughez) [inline docs/adding-a-resource.md:86]: Establish

File: docs/removing-a-resource.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  tag:  "Wellbeing",
+  details: {
+    resource_type: "Template",
+    reading_time: "4 minutes",
+    created_by: "Notre Dame High School"
+  }
+)%>
+```
+
+Delete the entire render block between the `<%=` and `%>`.
```
Comment (claire-hughez) [inline docs/removing-a-resource.md:32]: render?

File: docs/set-up.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
+# Set up
+
+## Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- Ruby (version 2.7 or higher)
```
Comment (claire-hughez) [inline docs/set-up.md:7]: i don't have any of these! D:

**Comment by claire-hughez** [comment]
These look fab. 

A couple questions I had:

1. Should we give some guidance on how PRs should be reviewed? I think this would be asking a dev from Claim additional payments to review, and someone in policy?
2. Depending on the trickiness of changing the colours, I wonder if we could also just have the page templates for the different sections ready for people to choose from? e.g. copy this one for wellbeing-resource-template, this one for pink identify-resource-template. I expect we'll need to do this for the presentation/resources with download template anyway.
3. In the context of when we’ll need this in the future, it makes sense to have 1 PR per resource as I imagine usually we won’t be adding that many. For now, should we do PRs 1:1 with resources, or can we open a PR where say a whole section is added?

File: docs/example-resource.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
+---
+title: INSERT-TITLE
+layout: /markdown_resource.html.erb
+---
+
+# INSERT-TITLE
+
+<strong class="govuk-tag">INSERT-RESOURCE-NAME</strong>
```
Comment (claire-hughez) [inline docs/example-resource.md:8]: Sorry, think this was me Abi. This should read 'INSERT-RESOURCE-TYPE'

File: docs/example-resource.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
+---
+title: INSERT-TITLE
+layout: /markdown_resource.html.erb
+---
+
+# INSERT-TITLE
```
Comment (claire-hughez) [inline docs/example-resource.md:6]: question from a n00b - does it matter if there's a space between the hash and the title or not?

**Comment by AbigailMcP** [comment]
> 1. Should we give some guidance on how PRs should be reviewed? I think this would be asking a dev from Claim additional payments to review, and someone in policy?

Done! Added to end of instructions

> 2. Depending on the trickiness of changing the colours, I wonder if we could also just have the page templates for the different sections ready for people to choose from? e.g. copy this one for wellbeing-resource-template, this one for pink identify-resource-template. I expect we'll need to do this for the presentation/resources with download template anyway.

This should just be a case of adding a new class to the info-box. Will update the instructions when that's merged.

> 3. In the context of when we’ll need this in the future, it makes sense to have 1 PR per resource as I imagine usually we won’t be adding that many. For now, should we do PRs 1:1 with resources, or can we open a PR where say a whole section is added?

To discuss.

File: docs/example-resource.md (hunk 10 above)
Comment (claire-hughez) [reply docs/example-resource.md:6]: omg that sucks, ive been doing that completely wrong :(

Comment (AbigailMcP) [reply docs/example-resource.md:6]: Actually scrap that, it seems to render correctly still! I think it's just best practice to add spacing: https://google.github.io/styleguide/docguide/style.html#add-spacing-to-headings

**Review by claire-hughez — APPROVED** [review]
woo!

**Pull request merged**